File: packages/appwsgi/appwsgi%20805.tar.bz2/trunk/www/vi/vi.py
# Copyright(c) gert.cuykens@gmail.com

from __future__ import with_statement
from xmlframe import read

def application(environ, response):
    xml = environ['wsgi.input'].read(int(environ.get('CONTENT_LENGTH','0')))
    v   = read(xml)

    if  v['vi'] == 'write':
        f = open('test.txt', 'w')
        f.write(v['txt'])
        f.close()

    xml  = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
    xml += "<root>"
    xml += "<![CDATA["

    with open('test.txt', 'r') as f:
         for line in f:
             xml += line

    xml += "]]>"
    xml += "</root>"

    response('200 OK',[('Content-type','text/xml'),('Content-Length', str(len(xml)))])
    return [xml]

File: packages/appwsgi/appwsgi%20667.tar.bz2/trunk/www/vi/vi.js
// Copyright(c) gert.cuykens@gmail.com
vi=
{
 'read':function()
 {
  http.fml=function(xml)
  {
   if (xml.getElementsByTagName('error')[0]) alert(xml.getElementsByTagName('error')[0].childNodes[0].nodeValue)
   document.getElementsByTagName('textarea')[0].value=xml.getElementsByTagName('root')[0].childNodes[0].data
  }
  http.xml ='<?xml version="1.0" encoding="UTF-8"?>\n'
  http.xml+='<root>\n'
  http.xml+=' <vi>read</vi>\n'
  http.xml+=' <txt>'+document.getElementsByTagName('textarea')[0].value+'</txt>\n'
  http.xml+='</root>'
  http.send()
 },

 'write':function()
 {
  http.fml=function(xml)
  {
   if (xml.getElementsByTagName('error')[0]) alert(xml.getElementsByTagName('error')[0].childNodes[0].nodeValue)
   document.getElementsByTagName('textarea')[0].value=xml.getElementsByTagName('root')[0].childNodes[0].data
  }
  http.xml ='<?xml version="1.0" encoding="UTF-8"?>\n'
  http.xml+='<root>\n'
  http.xml+=' <vi>write</vi>\n'
  http.xml+=' <txt>'+document.getElementsByTagName('textarea')[0].value+'</txt>\n'
  http.xml+='</root>'
  http.send()
 }
}

